langchain4j-core/src/main/java/dev/langchain4j/model/chat/request/ChatRequest.java

```diff
@@ -18,12 +18,13 @@ public class ChatRequest {
 
     private final List<ChatMessage> messages;
     private final List<ToolSpecification> toolSpecifications;
-    // TODO tool mode
+    private final ToolMode toolMode; // TODO wrap into "tools" pojo?
     private final ResponseFormat responseFormat;
 
     private ChatRequest(Builder builder) {
         this.messages = new ArrayList<>(ensureNotEmpty(builder.messages, "messages"));
         this.toolSpecifications = copyIfNotNull(builder.toolSpecifications);
+        this.toolMode = builder.toolMode; // TODO set AUTO by default? only if toolSpecifications are present?
```

Comment:
👍  on using `AUTO` by default if tool specs are present

Reply:
Hmm, usually we follow the "do not specify default values for optional parameters" strategy, so perharps we should do the same here? AFAIK, `AUTO` is the default value for all LLM providers, so `AUTO` will be set in any case